src/views/dialogs/ReferralDialog.tsx

```diff
@@ -37,18 +39,19 @@ const CONTENT_SECTIONS = [
 
 export const ReferralDialog = ({ setIsOpen, refCode }: DialogProps<ReferralDialogProps>) => {
   const stringGetter = useStringGetter();
+  const dispatch = useAppDispatch();
   const { dydxAddress } = useAccounts();
-  const { data: referralAddress, isFetched } = useReferralAddress(refCode);
-  const [, saveLastestReferrer] = useLocalStorage<string | undefined>({
-    key: LocalStorageKey.LatestReferrer,
-    defaultValue: undefined,
-  });
+  const { data: referralAddress, isSuccess } = useReferralAddress(refCode);
+  const { data: affiliatesInfo, isSuccess: isAffiliatesInfoSuccess } =
+    useAffiliatesInfo(referralAddress);
 
   useEffect(() => {
     if (referralAddress) {
-      saveLastestReferrer(referralAddress);
+      dispatch(updateLatestReferrer(referralAddress));
     }
-  }, [referralAddress, saveLastestReferrer]);
+  }, [dydxAddress, referralAddress, dispatch]);
```

Comment:
hmm why does dydxAddress have to be a dep here 👀

Reply:
hmmm good point, originally was thinking it needed update if account switch but, I don't think that make sense, will remove